src/Corax/IndexWriter.cs

```diff
@@ -304,6 +304,15 @@ void AddMaybeAvoidDuplicate(List<long> term, long entryId)
 
             void Insert(ReadOnlySpan<byte> value)
             {
+                if (value.Length > _encodingBufferHandler.Length)
+                {
+                    Analyzer.BufferPool.Return(_encodingBufferHandler);
+                    Analyzer.TokensPool.Return(_tokensBufferHandler);
+
+                    _encodingBufferHandler = Analyzer.BufferPool.Rent(value.Length);
+                    _tokensBufferHandler = Analyzer.TokensPool.Rent(value.Length);
+                }
+
```

Comment:
This is not exactly correct.  This functionality was provided in RavenDB-18820

Reply:
Does it mean we don't need this change at all?

Reply:
This is done here: https://github.com/ravendb/ravendb/blob/e66835161f42ab039fdaf2435ac86a4a7ae98130/src/Corax/IndexWriter.cs#L322-L334

Reply:
Then why it triggers that assertion?

Reply:
Because it is executed before checking inside `AnalyzeInsert`?

Reply:
I'm not sure what you mean. Those buffers are only for analyzers so they are changed only inside `AnalyzeInsert`.

Reply:
@redknightlois you created https://issues.hibernatingrhinos.com/issue/RavenDB-18872 for this right?